extension/bootstrap.js

```diff
@@ -78,20 +74,18 @@ var PersonasPlusBootstrapAddon = {
         Cu.unload("resource://personas/modules/personas.js");
         this.requestRemovePrerequisites(data);
         if (reason == ADDON_DISABLE || reason == ADDON_UNINSTALL) {
-            // modules that come with Firefox
-            Cu.import("resource://gre/modules/XPCOMUtils.jsm");
-            // LightweightThemeManager may not be not available (Firefox < 3.6 or Thunderbird)
-            try {
-                Cu.import("resource://gre/modules/LightweightThemeManager.jsm");
-            } catch (e) {
-                LightweightThemeManager = null;
-            }
-            try {
-                LightweightThemeManager.forgetUsedTheme(LightweightThemeManager.currentTheme.id);
-            } catch (e) {
-                try {
-                    LightweightThemeManager.currentTheme = null;
-                } catch (e) {}
+            if(LightweightThemeManager.currentTheme){
+                var prompt = Components.classes["@mozilla.org/embedcomp/prompt-service;1"].getService(Components.interfaces.nsIPromptService);
+                if (prompt.confirm(null, "Personas Plus", "Would you like to remove the current theme?")) {
+                    try {
+                        Cu.import("resource://gre/modules/LightweightThemeManager.jsm");
+                        LightweightThemeManager.forgetUsedTheme(LightweightThemeManager.currentTheme.id);
```

Comment:
Line 245 is the actual uninstall case. When complete uninstall happens, Groovy Blue is removed, whether it is active or disabled. In issue, you commented as:

> "If Groovy Blue is not the current theme, let's remove it anyway."

So I do not check whether it is active or not, just Groovy Blue is removed for uninstall case. It is default theme and should be removed if add-on is removed.

Otherwise there are too many combinations to handle which are not necessary for uninstall/disable events. If she wants to keep Groovy Blue, she can manually download from AMO.

BTW, LightweightThemeManager.forgetUsedTheme does not throw error, if the theme id is not available. It just silently fails.